ssl/ssl_test.cc

```diff
@@ -4597,6 +4805,10 @@ TEST(SSLTest, EmptyCipherList) {
   EXPECT_FALSE(SSL_CTX_set_cipher_list(ctx.get(), ""));
   ERR_clear_error();
 
+  // Configuring the empty cipher list fails.
+  EXPECT_FALSE(SSL_CTX_set_ciphersuites(ctx.get(), ""));
```

Comment:
This impl is intended and better IMO.

`|SSL_CTX_set_cipher_list/ciphersuites|` should tell nothing matched given not proper rule strs.

Reply:
I'm not sure I see a clear advantage of permitting empty lists or not. So, I would just align with 3.0.

Reply:
In aws-lc, `|SSL_CTX_set_cipher_list|` [does not allow empty lists, either](https://github.com/awslabs/aws-lc/blob/123eaaef26abc278f53ae338e9c758eb01c70b08/ssl/ssl_cipher.cc#L1270-L1280). If this alignment is preferred, create another SIM to track that.

Reply:
AFAIK `SSL_CTX_set_cipher_list()` doesn't allow an empty input in upstream either https://github.com/openssl/openssl/blob/master/ssl/ssl_lib.c#L2743-L2747.

Reply:
> I'm not sure I see a clear advantage of permitting empty lists or not. So, I would just align with 3.0.

The advantage is the API returns failure when no ciphers get matched given improper rule strs. I may revisit this part later because I vaguely remember MySQL8.0 may reply on `An empty list is permissible`.

Reply:
It still doesn't follow OpenSSL 3.0. And the other "TLS 1.2" specific function `SL_CTX_set_cipher_list()` does.